src/ProvidedTypes.fs

```diff
@@ -1554,8 +1554,23 @@ namespace ProviderImplementation.ProvidedTypes
                 |> Array.choose (function :? Type as m when memberBinds true bindingFlags false m.IsPublic || m.IsNestedPublic -> Some m | _ -> None)
                 |> (if hasFlag bindingFlags BindingFlags.DeclaredOnly || this.BaseType = null then id else (fun mems -> Array.append mems (this.ErasedBaseType.GetNestedTypes(bindingFlags)))))
 
-        override this.GetConstructorImpl(bindingFlags, _binder, _callConventions, _types, _modifiers) = 
-            let xs = this.GetConstructors bindingFlags |> Array.filter (fun m -> m.Name = ".ctor")
+        override this.GetConstructorImpl(bindingFlags, _binder, _callConventions, types, _modifiers) = 
+            let xs = 
+                this.GetConstructors bindingFlags 
+                |> Array.filter (fun m -> m.Name = ".ctor")
+                |> Array.filter 
+                    (fun m -> 
+                        if m.Name = ".ctor" then 
+                            let parameters = m.GetParameters() 
+                            if parameters.Length = types.Length then
+                                parameters 
+                                |> Seq.zip types 
+                                |> Seq.exists (fun (t,p) -> p.ParameterType <> t) 

```

Comment:
I'm concerned about this equality check on types.

Also, if we add this for constructors we should really add it for resolving method overloading as well.

https://github.com/fsprojects/FSharp.TypeProviders.SDK/pull/316/files#diff-7c540ae6a67f14c48d0403e7b021fc86R1588

Either way we will need tests

Reply:
I could see the equality check on types causing issues so I held off on tests.

At this point It could just be a check on the type name since `convTypeRef` makes a unique type name assumption anyway. I'm hesitant though, ideally `convTypeRef` could be updated to more accurately find types. Not sure if that's feasible...  Would a shallow equality check / hash on name/member names/parameter names/parameter type names be enough for now? Doesn't seem too elegant and would still leave some corner cases... Maybe simply keeping it at type name is good enough.

Reply:
Is performance the concern here? I thought Type.Equals was essentially just a reference comparison.